pywhat/Data/regex.json

```diff
@@ -249,10 +304,10 @@
     },
     {
        "Name": "YouTube Video ID",
-       "Regex": "^[0-9A-Za-z_-]{10}[048AEIMQUYcgkosw]{1}$",
+       "Regex": "^(?=.*[0-9])(?=.*[A-Z])(?=.*[a-z])[0-9A-Za-z_-]{10}[048AEIMQUYcgkosw]{1}$",

```

Comment:
Fun fact: 

I clicked a bunch of videos on my homepage, the 3rd video's URL was this:
https://www.youtube.com/watch?v=zKOMLoBP_Xw

so `_` does appear!

I would drop this down to a 0.2 personally -- it keeps on showing up and is annoying 😄 

_Perhaps_ we can implement some checkers (in the future!) that run Luhn checks (For cards) and sees if the youtube video exists (see https://www.youtube.com/watch?v=dlkadkjsak )

But for now, just drop it 😄

Reply:
PS: We can use that video as a test :eyes:

Reply:
Something we should see with testing, if we test well enough -- if we drop this down to 0.2 does this break the assumptions in our tests? Does the program still work as expected, minus those glitches?

Also, can you make a test for this glitch? Anytime someone says "XYZ doesn't work" and you fix it, make a test.

So test this:
```
078-05-1120
```

And test to make sure youtube videos don't appear first or above SSNs by doing `ret[0]`.

That way our tests form around what we expect our program to do, and if we change something (like moving this regex up or down the popularity ratings) will it break the program? Will it break what we expected it to do?

Reply:
^^ This is also why I have tests that don't run and are skipped, it's better to have tests that don't run and skip then to have no tests at all. That way we know where the limits are and can work on fixing them! :-)